helix-view/src/view.rs

```diff
@@ -117,7 +117,7 @@ pub struct View {
     // two last modified docs which we need to manually keep track of
     pub last_modified_docs: [Option<DocumentId>; 2],
     /// used to store previous selections of tree-sitter objects
-    pub object_selections: Vec<Selection>,
+    pub object_selections: HashMap<DocumentId, Vec<Selection>>,
```

Comment:
Storing per document data on the view is not a good idea. If the document is closed that data is essentially leaked and it also preculdes you from updating data on edits. If we are going to be storing this per-view and per-document then this should be stored on the `Document` as ` HashMap<ViewId,..>` just like selections are and inserted cleared in `remove_view` and `ensure_view_init`